### PR TITLE
LibWeb: Clear active targets before invoking them

### LINKED ISSUE #2011: LibWeb: Crash on pixiv.net
Crash happens consistently when loading [https://www.pixiv.net/en/artworks/95834820](https://www.pixiv.net/en/artworks/95834820)

gdb backtrace:

```sh
Thread 1 "WebContent" received signal SIGSEGV, Segmentation fault.
Web::DOM::Document::broadcast_active_resize_observations ()
    at /home/aplefull/Repos/ladybird/Userland/Libraries/LibWeb/DOM/Document.cpp:5163
5163	        observer->active_targets().clear();
(gdb) bt
#0  Web::DOM::Document::broadcast_active_resize_observations () at /home/aplefull/Repos/ladybird/Userland/Libraries/LibWeb/DOM/Document.cpp:5163
#1  0x00007a5926f4fc16 in Web::HTML::EventLoop::update_the_rendering () at /home/aplefull/Repos/ladybird/Userland/Libraries/LibWeb/HTML/EventLoop/EventLoop.cpp:349
#2  0x00007a5926f51269 in AK::Function<void ()>::operator()() const () at /home/aplefull/Repos/ladybird/AK/Function.h:120
#3  Web::HTML::Task::execute () at /home/aplefull/Repos/ladybird/Userland/Libraries/LibWeb/HTML/EventLoop/Task.cpp:47
#4  0x00007a5926f5029d in Web::HTML::EventLoop::process () at /home/aplefull/Repos/ladybird/Userland/Libraries/LibWeb/HTML/EventLoop/EventLoop.cpp:176
#5  0x00007a592720dc59 in JS::SafeFunction<void ()>::operator()() const () at /home/aplefull/Repos/ladybird/Userland/Libraries/LibJS/SafeFunction.h:85
#6  operator() () at /home/aplefull/Repos/ladybird/Userland/Libraries/LibWeb/Platform/TimerSerenity.cpp:23
#7  call () at /home/aplefull/Repos/ladybird/AK/Function.h:187
#8  0x00007a59293a6522 in AK::Function<void ()>::operator()() const () at /home/aplefull/Repos/ladybird/AK/Function.h:120
#9  Core::Timer::timer_event () at /home/aplefull/Repos/ladybird/Userland/Libraries/LibCore/Timer.cpp:97
#10 0x00007a59293951bd in Core::EventReceiver::dispatch_event () at /home/aplefull/Repos/ladybird/Userland/Libraries/LibCore/EventReceiver.cpp:162
#11 0x00005c74c5cb8b9d in qt_timer_fired () at /home/aplefull/Repos/ladybird/Ladybird/Qt/EventLoopImplementationQt.cpp:227
#12 operator() () at /home/aplefull/Repos/ladybird/Ladybird/Qt/EventLoopImplementationQt.cpp:241
#13 call () at /usr/include/x86_64-linux-gnu/qt6/QtCore/qobjectdefs_impl.h:127
#14 call<QtPrivate::List<>, void> () at /usr/include/x86_64-linux-gnu/qt6/QtCore/qobjectdefs_impl.h:241
#15 impl () at /usr/include/x86_64-linux-gnu/qt6/QtCore/qobjectdefs_impl.h:408
#16 0x00007a5928f83d9b in ??? () at /lib/x86_64-linux-gnu/libQt6Core.so.6
#17 0x00007a5928f8ed6d in QTimer::timeout(QTimer::QPrivateSignal) () at /lib/x86_64-linux-gnu/libQt6Core.so.6
#18 0x00007a5928f76576 in QObject::event(QEvent*) () at /lib/x86_64-linux-gnu/libQt6Core.so.6
#19 0x00007a5928f38416 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /lib/x86_64-linux-gnu/libQt6Core.so.6
#20 0x00007a592907b7ab in QTimerInfoList::activateTimers() () at /lib/x86_64-linux-gnu/libQt6Core.so.6
#21 0x00007a5929133a29 in ??? () at /lib/x86_64-linux-gnu/libQt6Core.so.6
#22 0x00007a59289145b5 in ??? () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#23 0x00007a5928973717 in ??? () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#24 0x00007a5928913a53 in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#25 0x00007a59291315ef in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()

```

### LINKED ISSUE #2475: hsedesign.ru crashes Ladybird
### Summary

When I open hsedesign.ru in Ladybird, it crashes

### Operating system

macOS

### Steps to reproduce

1. Open Ladybird
2. Navigate to hsedesign.ru
3. Observe

### Expected behavior

It doesn't crash

### Actual behavior

It crashes

### URL for a reduced test case

hsedesign.ru

### HTML/SVG/etc. source for a reduced test case

```html
N/A
```

### Log output and (if possible) backtrace

```shell
2024-11-21 11:09:55.391 Ladybird[86654:1655926] +[IMKClient subclass]: chose IMKClient_Modern
2024-11-21 11:09:55.439 Ladybird[86654:1655926] +[IMKInputSession subclass]: chose IMKInputSession_Modern
71338.373 WebContent(86657): (js log) "ym init for" "hsedesign.ru"
71338.373 WebContent(86657): Unhandled JavaScript exception: [TypeError] https://hsedesign.ru/ is not a function (evaluated from '<object>.href')
71338.373 WebContent(86657):     at https://hsedesign.ru/:1:2287
    at <unknown>
    at https://hsedesign.ru/:1:1

71338.375 WebContent(86657): ResourceLoader: Filtered request to: "https://www.googletagmanager.com/gtm.js?id=GTM-NL8PKVH"
71338.375 WebContent(86657): ResourceLoader: Filtered request to: "https://www.googletagmanager.com/gtag/js?id=G-DPXM9GVYLS"
71338.375 WebContent(86657): ResourceLoader: Filtered request to: "https://mc.yandex.ru/metrika/tag.js"
71338.375 WebContent(86657): HTMLScriptElement: Refusing to run script because the element's result is null.
71338.375 WebContent(86657): HTMLScriptElement: Refusing to run script because the element's result is null.
71338.375 WebContent(86657): HTMLScriptElement: Refusing to run script because the element's result is null.
71339.118 WebContent(86657): FIXME: Unimplemented Navigator.maxTouchPoints
71339.288 WebContent(86657): Unhandled JavaScript exception: [ReferenceError] 'XPathResult' is not defined
71339.288 WebContent(86657):     at https://top-fwz1.mail.ru/js/dyn-goal-config.js?ids=3566743:5:337
    at https://top-fwz1.mail.ru/js/dyn-goal-config.js?ids=3566743:7:508

71339.737 WebContent(86657): Failed to parse font: Automatic format detection failed
71345.772 WebContent(86657): FIXME: Unimplemented IDL interface: 'ServiceWorkerContainerPrototype.ready'
71345.772 WebContent(86657): Unhandled JavaScript exception: [TypeError] Cannot access property "then" on undefined object "navigator.serviceWorker.ready"
71345.772 WebContent(86657):     at https://hsedesign.ru/brg/static/js/main.73902fbe.js:33706:21195
    at https://hsedesign.ru/brg/static/js/main.73902fbe.js:33706:21195
    at https://hsedesign.ru/brg/static/js/main.73902fbe.js:33706:21195

71345.773 WebContent(86657): FIXME: Unimplemented IDL interface: 'LocationPrototype.ancestorOrigins'
71345.773 WebContent(86657): FIXME: Unimplemented IDL interface: 'LocationPrototype.ancestorOrigins'
71345.773 WebContent(86657): (js log) "StartRoutingMiniApps!!!" "hsedesign.ru" Location{ "href": "https://hsedesign.ru/", "origin": "https://hsedesign.ru", "protocol": "https:", "host": "hsedesign.ru", "hostname": "hsedesign.ru", "port": "", "pathname": "/", "search": "", "hash": "", "ancestorOrigins": undefined, "reload": [NativeFunction] , "replace": [NativeFunction] , "assign": [NativeFunction] , "toString": [NativeFunction]  } "/"
71345.774 WebContent(86657): (js log) "Приложение успешно найдено" Object{ "host": "hsedesign.ru", "mode": "production", "application": "deziiign/hsedesign" } Object{ "authentication": Object{ "App": [Function] App, "manifest": Object{ "title": "Burgundy Authorization", "reducer": Object{}, "auth": Object{ "productionClientId": "", "developClientId": "" }, "clear": true } }, "team": Object{ "App": [Function] App, "manifest": Object{ "title": "Team", "reducer": Object{}, "theme": Object{ "accent": Object{ "colorA": "", "colorB": "", "colorC": "", "colorD": "" }, "app": Object{ "backgroundColor": "#2A3034", "text": "#FEFEFE", "ten": "#545453" }, "tabs": Object{ "gradient": "#2A3034" }, "peoplePage": Object{ "title": Object{ "color": "#fff" } }, "mainSearchBar": Object{ "bg": "#3F4548", "placeholder": Object{ "color": "#909090" }, "input": Object{ "bg": "#3F4548", "placeholder": Object{ "color": "#575B5E" } }, "text": Object{ "color": "#FEFEFE", "hover": Object{ "color": "#353A3E" } }, "tabs": Object{ "color": "#909090", "colorActive": "#fefefe" } }, "longreadTranslated": Object{ "bg": "rgba(144, 144, 144, 0.8)", "color": "rgba(254, 254, 254, 1)" }, "mainSearchBarFooter": Object{ "bg": "#2A3034", "placeholder": Object{ "color": "#909090" } }, "educationSearchBar": Object{ "result": Object{ "color": "#fff" } }, "mainDropdownStyles": Object{ "colorActive": "#151514", "color": "#fefefe", "bg": "#3F4548", "hoverBG": "#575B5E" }, "mainBg": "#2A3034", "loading": "#8E9193", "tagView": Object{ "titleColor": "#FEFEFE", "goToMediaColor": "#FEFEFE", "goToMediaHoverColor": "#6C7073" }, "longread": Object{ "border": "none", "fontSize": "", "lineHeight": "" }, "progressbar": Object{ "color": "#8E9193", "colorBold": "#FEFEFE", "borderColor": "#6C7073" } undefined
71345.928 WebContent(86657): (js info) "[UserManager] getUser:" "user not found in storage"
71346.027 WebContent(86657): No property (from 2 properties) matched Token: Ident: px
71346.027 WebContent(86657):     row-gap
71346.027 WebContent(86657):     column-gap
71346.063 WebContent(86657): (js log) "location found"
71346.063 WebContent(86657): (js log) "location found"
71346.063 WebContent(86657): (js log) "location found"
71346.063 WebContent(86657): (js log) "location found"
71346.063 WebContent(86657): (js log) "location found"
71346.063 WebContent(86657): (js log) "location found"
71346.063 WebContent(86657): (js log) "location found"
71346.137 WebContent(86657): Failed to parse font: Automatic format detection failed
71346.137 WebContent(86657): Failed to parse font: Automatic format detection failed
71346.187 WebContent(86657): FIXME: Implement Audio::connect(AudioNode)
71346.187 WebContent(86657): FIXME: Implement Audio::connect(AudioNode)
71346.196 WebContent(86657): FIXME: Unimplemented IDL interface: 'CanvasRenderingContext2DPrototype.globalCompositeOperation'
71346.223 WebContent(86657): FIXME: Unimplemented Navigator.maxTouchPoints
71346.223 WebContent(86657): FIXME: Unimplemented Navigator.maxTouchPoints
71346.418 WebContent(86657): (js log) "Uploader initialization..."
71346.460 WebContent(86657): (js log) "tst" undefined false
71346.632 Ladybird(86654): WebContent process crashed! Last page loaded: https://hsedesign.ru/
71346.632 Ladybird(86654): Consider raising an issue at https://github.com/LadybirdBrowser/ladybird/issues/new/choose
71362.706 WebContent(86684): ConnectionBase::send_thread: send: Broken pipe (errno=32)
71362.932 Ladybird(86654): mach_msg failed: (ipc/rcv) port changed
```

### Screenshots or screen recordings

_No response_

### Build flags or config settings

_No response_

### Contribute a patch?

- [ ] I’ll contribute a patch for this myself.

### PR DESCRIPTION
**Pull request by ChaseKnowlden**
Fixes crash on hsedesign.ru.

Resolve #2475

### CODE REVIEW

File: Libraries/LibWeb/DOM/Document.cpp [hunk 1]
```diff
@@ -5203,11 +5203,11 @@ size_t Document::broadcast_active_resize_observations()
...
         }
 
-        // 4. Invoke observer.[[callback]] with entries.
-        observer->invoke_callback(entries);
-
-        // 5. Clear observer.[[activeTargets]].
+        // 4. Clear observer.[[activeTargets]].
         observer->active_targets().clear();
+
+        // 5. Invoke observer.[[callback]] with entries.
```
Comment (nico) [inline Libraries/LibWeb/DOM/Document.cpp:5209]: The lhs is the order in the spec: https://drafts.csswg.org/resize-observer-1/#broadcast-resize-notifications-h

If this no longer matches the spec, it probably should have at least a comment on why, and if this is the correct change, a spec bug too?

Comment (rmg-x) [reply Libraries/LibWeb/DOM/Document.cpp:5209]: The crash alone doesn't necessarily indicate there is a spec bug, especially with Ladybird being in a pre-alpha state.
I would probably investigate further and gain a more solid foundation / justification for why it needs to be this way (for us and the CSS wg)

**Comment by shlyakpavel** [comment]
This actually fixes https://github.com/LadybirdBrowser/ladybird/issues/2475

**Comment by ChaseKnowlden** [comment]
Closing.

**Pull request closed without merging**